go-controller/pkg/util/ovs.go

```diff
@@ -125,6 +125,67 @@ type execHelper struct {
 
 var runner *execHelper
 
+type ExecUtilRunSvc interface {
+	RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error)
+	Run(cmdPath string, args ...string) (string, string, error)
+	RunWithEnvVars(cmdPath string, envVars []string, args ...string) (string, string, error)
+}
+
+// ExecUtilRunSvcImplStruct implements the methods defined in the ExecUtilRunSvc interface
+type ExecUtilRunSvcImplStruct struct {
+}
+
+// RunCmd invokes the methods of the Cmd interfaces defined in k8s.io/utils/exec to execute commands
+// Note: the cmdPath and args parameter are used only for logging and is not processed
+func (runsvc *ExecUtilRunSvcImplStruct) RunCmd(cmd kexec.Cmd, cmdPath string, envVars []string, args ...string) (string, string, error) {
+	if cmd == nil {
+		return "", "", fmt.Errorf("cmd object cannot be nil")
+	}
+	if len(envVars) != 0 {
+		klog.V(5).Info("setting cmdSetEnv")
+		cmd.SetEnv(envVars)
+	}
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	cmd.SetStdout(stdout)
+	cmd.SetStderr(stderr)
+
+	counter := atomic.AddUint64(&runCounter, 1)
+	logCmd := fmt.Sprintf("%s %s", cmdPath, strings.Join(args, " "))
+	klog.V(5).Infof("exec(%d): %s", counter, logCmd)
+
+	err := cmd.Run()
+	klog.V(5).Infof("exec(%d): stdout: %q", counter, stdout)
+	klog.V(5).Infof("exec(%d): stderr: %q", counter, stderr)
+	if err != nil {
+		klog.V(5).Infof("exec(%d): err: %v", counter, err)
+	}
+	return strings.Trim(strings.TrimFunc(stdout.String(), unicode.IsSpace), "\" \n"),
```

Comment:
the old code didn't trim out these characters, why are you adding that now?I'm just wondering if other callers today do their own trimming, and expect to break output over multiple new lines

Reply:
As part of the refactor, I had to consider the below occurrences and converge them
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L271
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L279
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L287
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L301
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L415
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L427
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L469
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L475
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L482
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L493
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L505
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L517
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L535
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L587
- https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/ovs.go#L599

Reply:
In the latest PR pushed,I have reverted back to the previous behavior where the high level functions do the trimming instead of trimming happening in one single function.